internal/services/recoveryservices/site_recovery_replication_policy_hyperv_resource.go

```diff
@@ -0,0 +1,244 @@
+	RecoveryPointRetentionInHours                 int64  `tfschema:"recovery_point_retention_in_hours"`
+	ApplicationConsistentSnapshotFrequencyInHours int64  `tfschema:"application_consistent_snapshot_frequency_in_hours"`
+	CopyFrequency                                 int64  `tfschema:"replication_interval_in_seconds"`
+}
+
+type ReplicationPolicyHyperVResource struct{}
+
+var _ sdk.ResourceWithUpdate = ReplicationPolicyHyperVResource{}
+
+func (r ReplicationPolicyHyperVResource) Arguments() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+		"resource_group_name": commonschema.ResourceGroupName(),
+
+		"recovery_vault_name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validate.RecoveryServicesVaultName,
+		},
+		"recovery_point_retention_in_hours": {
+			Type:         pluginsdk.TypeInt,
+			Required:     true,
+			ForceNew:     false,
+			ValidateFunc: validation.IntBetween(0, 24),
+		},
+		"application_consistent_snapshot_frequency_in_hours": {
+			Type:         pluginsdk.TypeInt,
+			Required:     true,
+			ForceNew:     false,
+			ValidateFunc: validation.IntBetween(0, 12),
+		},
+		"replication_interval_in_seconds": {
+			Type:         pluginsdk.TypeInt,
+			Required:     true,
+			ForceNew:     false,
+			ValidateFunc: validation.IntInSlice([]int{30, 300}),
+		},
+	}
+}
+
+func (r ReplicationPolicyHyperVResource) Attributes() map[string]*schema.Schema {
+	return map[string]*schema.Schema{}
+}
+
+func (r ReplicationPolicyHyperVResource) ModelObject() interface{} {
+	return &ReplicationPolicyHyperVModel{}
+}
+
+func (r ReplicationPolicyHyperVResource) ResourceType() string {
+	return "azurerm_site_recovery_replication_policy_hyperv"
+}
+
+func (r ReplicationPolicyHyperVResource) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			var plan ReplicationPolicyHyperVModel
+			if err := metadata.Decode(&plan); err != nil {
+				return fmt.Errorf("decoding %+v", err)
+			}
+
+			subscriptionId := metadata.Client.Account.SubscriptionId
+			client := metadata.Client.RecoveryServices.ReplicationPoliciesClient
+
+			id := replicationpolicies.NewReplicationPolicyID(subscriptionId, plan.ResourceGroupName, plan.RecoveryVaultName, plan.Name)
+
+			existing, err := client.Get(ctx, id)
+			if err != nil {
+				// NOTE: Bad Request due to https://github.com/Azure/azure-rest-api-specs/issues/12759
+				if !response.WasNotFound(existing.HttpResponse) && !wasBadRequestWithNotExist(existing.HttpResponse, err) {
+					return fmt.Errorf("checking presence %s: %+v", plan.Name, err)
+				}
+			}
+
+			if existing.Model != nil && existing.Model.Id != nil && *existing.Model.Id != "" {
```

Comment:
It's sufficient to just check whether the Model exists in this case
```suggestion
			if existing.Model != nil {
```